lib/created_message_queue_handler.ts

```diff
@@ -100,71 +110,95 @@ export async function handleMessage(
     retrievedMessage.fiscalCode
   );
 
-  if (errorOrMaybeProfile.isRight) {
-    // query succeeded, we may have a profile
-    const maybeProfile = errorOrMaybeProfile.right;
-
-    //
-    // attempt to resolve an email notification
-    //
-    const maybeProfileEmail = maybeProfile
-      .flatMap(profile => option(profile.email))
-      .map(email => Tuple2(email, NotificationAddressSource.PROFILE_ADDRESS));
-    const maybeDefaultEmail = defaultAddresses
-      .flatMap(addresses => option(addresses.email))
-      .map(email => Tuple2(email, NotificationAddressSource.DEFAULT_ADDRESS));
-    const maybeEmail = maybeProfileEmail.orElse(() => maybeDefaultEmail);
-    const maybeEmailNotification: Option<
-      INotificationChannelEmail
-    > = maybeEmail.map(({ e1: toAddress, e2: addressSource }) => {
-      return {
-        addressSource,
-        status: NotificationChannelStatus.QUEUED,
-        toAddress
-      };
-    });
-
-    // check whether there's at least a channel we can send the notification to
-    if (maybeEmailNotification.isDefined) {
-      // create a new Notification object with the configured notifications
-      const notification: INewNotification = {
-        emailNotification: maybeEmailNotification.isDefined
-          ? maybeEmailNotification.get
-          : undefined,
-        fiscalCode: retrievedMessage.fiscalCode,
-        id: toNonEmptyString(ulid()).get,
-        kind: "INewNotification",
-        messageId: retrievedMessage.id
-      };
-
-      // save the Notification
-      const result = await notificationModel.create(
-        notification,
-        notification.messageId
-      );
+  if (errorOrMaybeProfile.isLeft) {
+    // query failed
+    return left(ProcessingError.TRANSIENT);
+  }
 
-      if (result.isRight) {
-        // save succeeded, return the saved Notification
-        return right(result.right);
-      } else {
-        // saved failed, fail with a transient error
-        // TODO: we could check the error to see if it's actually transient
-        return left(ProcessingError.TRANSIENT);
+  // query succeeded, we may have a profile
+  const maybeProfile = errorOrMaybeProfile.right;
+
+  // whether the recipient wants us to store the message content
+  const isMessageStorageEnabled =
+    maybeProfile
+      .map(profile => profile.isStorageOfMessageContentEnabled)
+      .getOrElse(false) || false;
```

Comment:
why || false ?

Reply:
`isStorageOfMessageContentEnabled` can be `undefined`, so `getOrElse(false)` returns `false | undefined`, so: the first `false` is when there's no profile, the 2nd `false` is when there's a profile but `isStorageOfMessageContentEnabled` is undefined.